membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -70,4 +82,35 @@ class AttributeController extends Controller with LazyLogging {
 
   def membership = lookup("membership", identity[Attributes])
   def features = lookup("features", onSuccess = Features.fromAttributes, onNotFound = Some(Features.unauthenticated))
+  
+  def updateAttributes(identityId : String): Action[AnyContent] = backendForSyncWithZuora.async { implicit request =>
+
+    val tp = request.touchpoint
+
+    val result: EitherT[Future, String, Attributes] =
+      for {
+        contact <- EitherT(tp.contactRepo.get(identityId).map(_ \/> s"No contact for $identityId"))
+        memSubF = EitherT[Future, String, Option[Subscription[SubscriptionPlan.Member]]](tp.subService.current[SubscriptionPlan.Member](contact).map(a => \/.right(a.headOption)))
+        conSubF = EitherT[Future, String, Option[Subscription[SubscriptionPlan.Contributor]]](tp.subService.current[SubscriptionPlan.Contributor](contact).map(a => \/.right(a.headOption)))
+        memSub <- memSubF
+        conSub <- conSubF
+        _ <- EitherT(Future.successful(if (memSub.isEmpty && conSub.isEmpty) \/.left("No paying relationship") else \/.right(())))
+        attributes = Attributes( UserId = identityId, Tier = memSub.map(_.plan.charges.benefit.id), MembershipNumber = contact.regNumber, Contributor = conSub.map(_.plan.charges.benefit.id))
```

Comment:
The benefitId of a contributor currently is Contributor. I believe we should change it to monthly-contribution, so in the future, if we want to add annual or other type is easily extensible.

The API returns the following data for a member who is a contributor:
<img width="286" alt="picture 279" src="https://user-images.githubusercontent.com/825398/27186009-00abdbae-51df-11e7-8fef-12fb2a188efb.png">